src/main/service/subscriptionService.ts

```diff
@@ -66,4 +73,119 @@ export class SubscriptionService {
+          caseDetailsList = await this.getCaseDetails(hearingIdsList);
+          // set results into cache
+          await this.setPendingSubscriptions(caseDetailsList, 'cases', user.id);
+          break;
+        case 'court-selections[]':
+          Array.isArray(pendingSubscription[`${selectionName}`]) ?
+            courtIdsList = pendingSubscription[`${selectionName}`] :
+            courtIdsList.push(pendingSubscription[`${selectionName}`]);
+          courtDetailsList = await this.getCourtDetails(courtIdsList);
+          // set results into cache
+          await this.setPendingSubscriptions(courtDetailsList, 'courts', user.id);
+          break;
+      }
+    }
+  }
+
+  public async getCaseDetails(cases): Promise<Hearing[]> {
+    const casesList = [];
+    for (const caseNumber of cases) {
+      const caseDetails = await hearingService.getCaseByNumber(caseNumber);
+      if (caseDetails) {
+        casesList.push(caseDetails);
+      }
+    }
+    return casesList;
+  }
+
+  public async getCourtDetails(courts): Promise<Court[]> {
+    const courtsList = [];
+    for (const courtId of courts) {
+      const courtDetails = await courtService.getCourtById(courtId);
+      if (courtDetails) {
+        courtsList.push(courtDetails);
+      }
+    }
+    return courtsList;
+  }
+
+  public async setPendingSubscriptions(subscriptions, subscriptionType, userId): Promise<void> {
+    await pendingSubscriptionsFromCache.setPendingSubscriptions(subscriptions, subscriptionType, userId);
+  }
+
+  public async getPendingSubscriptions(userId, subscriptionType): Promise<any[]> {
+    return await pendingSubscriptionsFromCache.getPendingSubscriptions(userId, subscriptionType);
+  }
+
+  public async subscribe(userId): Promise<boolean> {
+    let subscribed = true;
+    const casesType = 'cases';
+    const courtsType = 'courts';
+    const cachedCaseSubs = await pendingSubscriptionsFromCache.getPendingSubscriptions(userId, casesType);
+    const cachedCourtSubs = await pendingSubscriptionsFromCache.getPendingSubscriptions(userId, courtsType);
+    if (cachedCaseSubs) {
+      for (const cachedCase of cachedCaseSubs) {
+        const response = await subscriptionRequests.subscribe(this.createSubscriptionPayload(cachedCase, casesType, userId));
+        response ? await this.removeFromCache({'case': cachedCase.caseNumber}, userId) : subscribed = response;
+      }
+    }
+    if (cachedCourtSubs) {
+      for (const cachedCourt of cachedCourtSubs) {
+        const response = await subscriptionRequests.subscribe(this.createSubscriptionPayload(cachedCourt, courtsType, userId));
+        response ? await this.removeFromCache({court: cachedCourt.courtId}, userId) : subscribed = response;
+      }
+    }
+    return subscribed;
+  }
+
+  createSubscriptionPayload(pendingSubscription, subscriptionType, userId): object {
+    let payload;
+    switch (subscriptionType) {
+      case 'courts':
+        payload = {
+          channel: 'API',
```

Comment:
How come this is API? should it not be Email?

Reply:
Was my misunderstanding, refactored to EMAIL